p2p/gossip.go

```diff
@@ -114,3 +116,12 @@ func (g *Gossiper) ProcessMessages(ctx context.Context) {
 		}
 	}
 }
+
+func wrapValidator(validator GossipValidator) pubsub.Validator {
+	return func(_ context.Context, _ peer.ID, msg *pubsub.Message) bool {
+		return validator(&GossipMessage{
+			Data: msg.Data,
+			From: msg.GetFrom(),
+		})
+	}
+}
```

Comment:
I don't yet understand the benefits of wrapping the pubsub types here. Why not keep the types as is? keeping it as a `pubsub.Validator` seems much less opaque to me.

Reply:
I wanted to create abstraction over libp2p. So validator function have no dependencies on libp2p, as validators (and handlers) are supposed to be "application logic", and shouldn't care about libp2p details. It should be also anti-fragile solution - for example in case of change of libp2p types (`pubsub.Validator` or `pubsub.Message`), we need to fix in one place, not in every validator.

But maybe it's just over-engineering ;)